src/renderer/webgl/shaders/LoaderShader.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import DefaultShader from "./DefaultShader.mjs";
+import StageUtils from "../../../tree/StageUtils.mjs";
+
+export default class LoaderShader extends DefaultShader {
```

Comment:
Could be renamed to SpinnerShader2 to indicate this is new and improved version, the old one can would remain for backwards compatibility.

Reply:
Ok changed it to Spinner2

Reply:
Would SpinnerShader2 be a public name? I.e. as an app developer you would use it in your code as `Lightning.shaders.SpinnerShader2` or something like that?

Reply:
Something like that, you would use `Lightning.shaders.Spinner2`.

Reply:
My 2 cents:

if possible we should just replace the old Spinner shader with the new version (Spinner2) if that one is better. I imagine it's possible to add some support for the old API (if there are any changes) to not break backward compatibility.

the only reason I'd keep 2 spinners side by side, is when Spinner2 isn't capable of doing something that Spinner1 could do. But in that case I'd choose a more descriptive name for the new shader than just appending '2'

Reply:
The main difference is that the old shader can be used on top of existing texture as overlay (all its shortcomings come from this feature), while new one doesn't have it but instead is overall better quality. Also, any changes that have impact on visuals of the existing apps I would consider backwards incompatible and this shader falls into that category. Spinner2 is not such a bad name, especially because of recently released lightning 2.0, since it could either mean new and improved version but also as a reference to new lightning version.

Reply:
I think the goal should be to have only one Spinner Shader. And if we need more, I’d really prefer not to call it Spinner2 if we can prevent that. 

My understanding so far is that “Spinner1” requires a background texture and “Spinner2” doesn’t (but also doesn’t display it when specified), right @jfboeve ?

So wouldn’t we achieve BC by just adding the functionality to display a background texture to the new Spinner?

Reply:
I can understand in the future more spinners could be added e.g. https://projects.lukehaas.me/css-loaders/ . The main visual difference between old and new one is that new one has circular tip. @michielvandergeest, perhaps you can suggest naming scheme. Also, trying to make the two compatible turned out to be not feasible so we're stuck with both (which is fine because by default they look different)